framework/wazuh/agent.py

```diff
@@ -546,8 +554,26 @@ async def add_agent(
     if len(name) > common.AGENT_NAME_LEN_LIMIT:
         raise WazuhError(1738)
 
-    async with get_indexer_client() as indexer:
-        new_agent = await indexer.agents.create(id=id, name=name, key=key, groups=groups, ips=ips, os=os)
+    async with get_indexer_client() as indexer_client:
+        new_agent = await indexer_client.agents.create(
+            id=id,
+            name=name,
+            key=key,
+            type=type,
+            version=version,
+            groups=','.join(groups),
```

Comment:
The groups are also inserted into the Agents index as an array of strings.

Reply:
This requires updating agent groups queries and scripts. I opened the issue https://github.com/wazuh/wazuh/issues/26847 to perform these changes.